mkctr.go

```diff
@@ -158,10 +160,29 @@ func fetchBaseImage(baseImage string, opts ...remote.Option) (*remote.Descriptor
 	return desc, nil
 }
 
+// canRunLocal reports whether the platform can run the binary locally, to be
+// used by the local target.
+func canRunLocal(p v1.Platform) bool {
+	if p.OS != "linux" {
+		return false
+	}
+	if runtime.GOOS == "linux" {
+		return p.Architecture == runtime.GOARCH
+	}
+	if runtime.GOOS == "darwin" {
+		// macOS can run amd64 linux binaries in docker.
+		return p.Architecture == "amd64"
+	}
+	return false
+}
+
 func verifyPlatform(p v1.Platform, target string) error {
 	if p.OS != "linux" {
 		return fmt.Errorf("unsupported OS: %v", p.OS)
 	}
+	if target == "local" && canRunLocal(p) {
+		return fmt.Errorf("not required for target %q", target)
```

Comment:
I'm not quite clear on what this error is telling me and what I would do about it if I saw it.  I've specified `--target=local`, but it's not required for the architecture I'm on?  Why would that be an error state?  (I'm sure I'm misunderstanding)